packages/react-html/README.md

```diff
@@ -12,112 +12,267 @@ $ yarn add @shopify/react-html
+##### `getSerialized<Data>()`
 
-## Asset Components
+To help in migration, this function can imperatively return the parsed value of a serialization. It returns the data cast to whatever is passed for `Data`. It should only be called on the client.
 
-This module also exports the asset components the `<HTML />` component uses internally for its script and style props.
+#### Asset components
 
-```ts
-import {Style, Script} from '@shopify/react-html';
-```
+The main entrypoint also exposes components for easily building links to your assets:
 
-### Style
+##### `<Style />`
 
-The `<Style />` component lets you render `<link>` tags in your document dynamically as part of your react app.
+The `<Style />` component lets you render `<link>` tags in your document dynamically as part of your react app. It supports all of the props of a basic `link` tag, but forces some properties to be the values needed for a stylesheet.
+
+```tsx
+import {Style} from '@shopify/react-html';
 
-```ts
 <Style
   href="./some-style.css"
   integrity="some-integrity-hash"
   crossOrigin="anonymous"
-/>
+/>;
 ```
 
-### Script
+##### `<Script />`
 
-The `<Script />` component lets you render `<script>` tags in your document dynamically as part of your react app.
+The `<Script />` component lets you render `<script>` tags in your document dynamically as part of your react app. It supports all the props of a basic `script` tag.
+
+```tsx
+import {Script} from '@shopify/react-html';
 
-```ts
 <Script
   src="./some-script.js"
   integrity="some-integrity-hash"
   crossOrigin="anonymous"
+/>;
+```
+
+### Server
+
+The server entrypoint allows you to easily construct an HTML document for the response. It also re-exports all of the components noted above.
+
+#### `<Html />` and `render()`
+
+The `<Html>` component serves as a top level wrapper for a React application, allowing you to avoid needing any kind of server-side template. The `render()` function creates a stringified version of the HTML document with an appropriate DOCTYPE.
+
+```tsx
+import * as React from 'react';
+import {render, Html} from '@shopify/react-html/server';
+
+import App from '../app';
+
+export default function middleware(ctx, next) {
+  ctx.body = render(
+    <Html>
+      <App />
+    </Html>,
+  );
+
+  await next();
+}
+```
+
+The component will automatically propagate any usage of the [`react-helmet` library](https://github.com/nfl/react-helmet) in your app’s content to manipulate the title or other top level HTML or HEAD attributes.
+
+##### Props
+
+The `<Html />` component accepts the following props:
+
+- `manager`: a `Manager` instance. When provided, the `Html` component will extract all the information from this object and place it in an appropriate place in the document.
+- `children`: the application. It will be rendered to a string and placed inside a `div` with an ID of `app`.
+- `locale`: the language to use for the HTML `lang` attribute.
```

Comment:
Why not call this prop `lang` so it matches the attribute that gets populated?

Reply:
It's an interesting idea, I chose it because the variable you are very likely to have in scope is `locale`, as this is used for the i18n library. IMO the `lang` attribute is more of an implementation detail of the HTML component.